content/blog/2022/2022-02-15-getting-started-with-spring-webflux.md

```diff
@@ -0,0 +1,566 @@
+        return ServerResponse
+                .ok()
+                .contentType(MediaType.APPLICATION_JSON)
+                .body(userService.getAllUsers(), User.class);
+    }
+
+    public Mono<ServerResponse> getUserById(ServerRequest request) {
+        return userService
+                .findById(request.pathVariable("userId"))
+                .flatMap(user -> ServerResponse
+                        .ok()
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .body(user, User.class)
+                )
+                .switchIfEmpty(ServerResponse.notFound().build());
+    }
+
+    public Mono<ServerResponse> create(ServerRequest request) {
+        Mono<User> user = request.bodyToMono(User.class);
+
+        return user
+                .flatMap(u -> ServerResponse
+                        .status(HttpStatus.CREATED)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .body(userService.createUser(u), User.class)
+                );
+    }
+
+    public Mono<ServerResponse> updateUserById(ServerRequest request) {
+        String id = request.pathVariable("userId");
+        Mono<User> updatedUser = request.bodyToMono(User.class);
+
+        return updatedUser
+                .flatMap(u -> ServerResponse
+                        .ok()
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .body(userService.updateUser(id, u), User.class)
+                );
+    }
+
+    public Mono<ServerResponse> deleteUserById(ServerRequest request){
+        return userService.deleteUser(request.pathVariable("userId"))
+                .flatMap(u -> ServerResponse.ok().body(u, User.class))
+                .switchIfEmpty(ServerResponse.notFound().build());
+    }
+}
+```
+
+Next, we will define the router function. Router functions usually evaluate the request and choose the appropriate handler function. They serve as an alternate to the *@RequestMapping* annotation. So we will define this *RouterFunction* and annotate as a Bean. Next we can add it to a config, most likely `RouterConfig`:
+
+```java
+@Configuration
+public class RouterConfig {
+
+    @Bean
+    RouterFunction<ServerResponse> routes(UserHandler handler) {
+        return route(GET("/handler/users").and(accept(MediaType.APPLICATION_JSON)), handler::getAllUsers)
+                .andRoute(GET("/handler/users/{userId}").and(contentType(MediaType.APPLICATION_JSON)), handler::getUserById)
+                .andRoute(POST("/handler/users").and(accept(MediaType.APPLICATION_JSON)), handler::create)
+                .andRoute(PUT("/handler/users/{userId}").and(contentType(MediaType.APPLICATION_JSON)), handler::updateUserById)
+                .andRoute(DELETE("/handler/users/{userId}").and(accept(MediaType.APPLICATION_JSON)), handler::deleteUserById);
+    }
+}
+```
+
+Finally, we will define some properties in order to configure our database connection and server config.
+
+```yaml
```

Comment:
Please mention that this is the `application.yml` file.

Reply:
I have added it in the sentence.